src/sedcli_main.c

```diff
@@ -654,6 +701,41 @@ static int handle_setpw(void)
 	return ret;
 }
 
+static int handle_mbr_control(void)
+{
+	struct sed_device *dev = NULL;
+	int ret;
+
+	sedcli_printf(LOG_INFO, "Enter Admin1 password: ");
+
+	ret = get_password((char *) opts->pwd.key, &opts->pwd.len,
+			SED_MIN_KEY_LEN, SED_MAX_KEY_LEN);
+	if (ret)
+		return -1;
+
+	ret = sed_init(&dev, opts->dev_path);
```

Comment:
[root@localhost src]# ./sedcli -D -d /dev/nvme1n1 

SED TPER FEATURES SUPPORTED
	Sync Supported        : Y
	Async Supported       : N
	ACK/NAK Supported     : N
	Buffer Mgmt Supported : N
	Streaming Supported   : Y
	ComID Mgmt Supported  : N

SED LOCKING FEATURES SUPPORTED
	Locking Supported : Y
	Locking Enabled   : N
	Locked            : N
	Media Encryption  : Y
	MBR Enabled       : N
	MBR Done          : N

SED GEOMETRY FEATURES SUPPORTED
	Alignment required    : N
	Logical Block Size    : 512
	Alignment Granularity : 1
	Lowest Aligned LBA    : 0

SED DATASTORE FEATURES SUPPORTED
	Max DataStore tables       : 10
	Max size DataStore tables  : 10485760
	DataStore table size align : 4096

SED Opal v1.00 FEATURES SUPPORTED
	Base ComID       : 2048
	Number of ComIDs : 1

SED Opal v2.00 FEATURES SUPPORTED
	Base ComID                      : 2048
	Number of ComIDs                : 1
	Range Crossing Behavior         : 1
	Admin Authorities LSP Supported : 4
	User Authorities LSP Supported  : 9
	Initial PIN                     : 0
	Revert PIN                      : 0

[root@localhost src]# ./sedcli -M -d /dev/nvme1n1 -e TRUE -m TRUE
Enter Admin1 password: 
Password too short..!!
Please provide password min 8 characters long.
[root@localhost src]# test
[root@localhost src]# ./sedcli -M -d /dev/nvme1n1 -e TRUE -m TRUE
Enter Admin1 password: 
Invalid Parameter

Reply:
# sedcli -M -d /dev/nvme0n1 -e TRUE -m TRUE
Enter Admin1 password:
Success

Works fine on my system...Can you please re-try again

Reply:
I tried on fresh drive from factory -- that's why I included log from discovery
Could you please try the same?

Reply:
You need to activate-lsp before you perform any of the MBR control operations.
Once the lsp is activated, the discover command should show 'Locking Enabled : Y'.
Any MBR control operations performed after this should succeed.

Reply:
Exactly. Checks should be added in SW making sure that drive is in correct state. And if not error should be returned to user indicating appropriate reason of failure

Reply:
I would say that the user should know that the LSP should be activated before the operation is performed. Also, we are providing a Discover Zero command under nvme_pt interface to assist the user with necessary info.
Also, we can provide a new cli to the user where ownership, activate lsp and setup global locking range are all done in 1 command (initial_setup)

Reply:
It is not obvious for user from printed out error (Invalid Parameter) to deduce what needs to be done (take ownership, activate LSP, etc). This needs to be fixed.

Providing aggregate cli command to do multiple things  is a good idea. I thin it is lower priority than MBR control & write because method for intializing device already exists (through multiple commands) while MBR control & write doens't